### PR TITLE
test: add unit tests for keyword modules

### PR DESCRIPTION
**Pull request by Dor-bl**
- Add tests for _keyevent.py (100% coverage)
- Add tests for _runonfailure.py (81% coverage)
- Add tests for _screenshot.py (100% coverage)
- Add tests for _screenrecord.py (57% coverage)
- Add tests for _waiting.py (100% coverage)

Improves overall test coverage from 33% to 39%

## Implements
This pull request adds comprehensive unit tests for several internal keyword classes in the AppiumLibrary, significantly improving test coverage for key functionalities such as key events, run-on-failure behavior, screen recording, screenshots, and waiting mechanisms. The new tests use Python's `unittest` framework and extensive mocking to verify correct behavior under various scenarios, including edge cases and error handling.

**New Unit Test Coverage:**

* **Key Event Keywords:**
  - Added tests for `_KeyeventKeywords`, covering `press_keycode` and `long_press_keycode` methods, including handling of meta states and string-to-int conversion.

* **Run-On-Failure Keywords:**
  - Introduced tests for `_RunOnFailureKeywords`, verifying registration, state transitions, disabling, and error reporting logic for run-on-failure keywords.

* **Screen Recording Keywords:**
  - Implemented tests for `_ScreenrecordKeywords`, checking start/stop behavior, time limits, output format handling, prevention of duplicate recordings, and error handling for missing sessions.

* **Screenshot Keywords:**
  - Added tests for `_ScreenshotKeywords`, ensuring correct screenshot capture with and without filenames, file saving, fallback mechanisms, subdirectory handling, and HTML embedding.

* **Waiting Keywords:**
  - Developed tests for `_WaitingKeywords`, covering sleep interval management, timeout formatting, and all major wait-until behaviors (element/text presence/absence), including error and edge cases.

### CODE REVIEW

**Review by chsinger — COMMENTED** [review]
Thanks for the PR @Dor-bl, solid test additions — I ran the suite on your branch and after a test-merge onto current master, all green. Two small suggestions below.

File: tests/keywords/test_waiting.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    def test_set_sleep_between_wait_loop(self):
+        """Test setting the sleep interval."""
+        old_value = self.wk.set_sleep_between_wait_loop(0.5)
+        self.assertEqual(old_value, 0.2)
+        self.assertEqual(self.wk._sleep_between_wait, 0.5)
+
+    def test_set_sleep_between_wait_loop_with_string(self):
+        """Test setting the sleep interval with a string value."""
+        self.wk.set_sleep_between_wait_loop(robot.utils.timestr_to_secs('1s'))
```
Comment (chsinger) [inline tests/keywords/test_waiting.py]: The string is already converted to a float here before the keyword is called, so the string-parsing path inside `set_sleep_between_wait_loop` isn't actually exercised (which is what the test name promises). Passing the raw string covers it:

```suggestion
        self.wk.set_sleep_between_wait_loop('1s')
```

File: tests/keywords/test_screenrecord.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    def setUp(self):
+        """Set up the test fixture."""
+        self.sr = _ScreenrecordKeywords()
+        self.mock_driver = mock.Mock()
+        self.sr._current_application = mock.Mock(return_value=self.mock_driver)
+        self.sr._html = mock.Mock()
+        self.sr._get_log_dir = mock.Mock(return_value='/tmp/logs')
+        # Mock platform detection
+        self.sr._set_output_format = mock.Mock(return_value='mp4')
```
Comment (chsinger) [inline tests/keywords/test_screenrecord.py]: The real `_set_output_format` returns `'.mp4'` (with a leading dot). Keeping the mock consistent avoids surprises if a later test asserts a generated filename:

```suggestion
        self.sr._set_output_format = mock.Mock(return_value='.mp4')
```

**Comment by Dor-bl** [comment]
Applied the suggestions! Thank you for the review.

**Review by chsinger — APPROVED** [review]
